File: projekt3/Guarana.py
from Plant import *

class Guarana(Plant):
    __GUARANA_STRENGTH__ = 0
    __STRENGTH_BONUS__ = 3
    __COLOR__ = "Red"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__GUARANA_STRENGTH__,world)
    
    def getColor(self):
        return self.__COLOR__
      
    def getName(self):
      return "Guarana"

    def spread(self, newRow, newCol):
      self._world_.setOrganism(newRow, newCol, Guarana(newRow , newCol, self._world_))

    def specialAbility(self, other):
        other.setExtraStrength(self.__STRENGTH_BONUS__)
        event = other.getName() + " zjada guaranne jego siła wzrasta o " + str(self.__STRENGTH_BONUS__)
        self._world_.addEvent(event)
        self._world_.removeOrganism(self.getRow(), self.getCol())
        self.die()
        return True
    

File: projekt2/src/proj/java/model/plant/Guarana.java
package proj.java.model.plant;

import java.awt.Color;

import proj.java.model.Organism;
import proj.java.model.Plant;
import proj.java.model.World;

public class Guarana extends Plant {

  private static final int GUARANA_STRENGTH = 0;
  private static final int STRENGTH_BONUS = 3;

  @Override
  protected void spread(int newRow, int newCol, World world) {
      world.setOrganism(newRow, newCol, new Guarana());
  }

  @Override
  public int getStrength() {
    return GUARANA_STRENGTH;
  }

  @Override
  public Color getColor() {
    return Color.RED;
  }

  @Override
  public boolean specialAbility(Organism other, World world, boolean specialAblityStopsMovement) {
    other.setExtraStrength(STRENGTH_BONUS);
    String message = String.format("%s zjada guaranne jego siła wzrasta o %d" , other.getName(), STRENGTH_BONUS);
    world.addMessage(message);
    world.removeOrganism(this.getRow(), this.getCol());
    specialAblityStopsMovement = false;
    return true;
  }
}
